File: java/src/ev3/ut6_pooavanzadaestructuras/u01herenciapolimorfismo/teoria/U03_Polimorfismo/Rectangulo.java
package ev3.ut6_pooavanzadaestructuras.u01herenciapolimorfismo.teoria.U03_Polimorfismo;

/**
 * Figura concreta: rectangulo.
 *
 * @author Agustin. A. Marquez. Pina
 * @since 18/09/2026
 * @see Figura
 * @see U03_PolimorfismoIntro
 * @see <a href="mailto:agu1406@outlook.es">agu1406@outlook.es</a>
 * @see <a href="https://github.com/Agu1406/ClasesParticulares">Repositorio GitHub</a>
 * @see <a href="https://www.agustinmarquez.dev">Sitio web</a>
 */
public class Rectangulo extends Figura {

    private double base;
    private double altura;

    public Rectangulo(double base, double altura) {
        this(base, altura, "negro");
    }

    public Rectangulo(double base, double altura, String color) {
        super("Rectangulo", color);
        this.base = base;
        this.altura = altura;
    }

    public double getBase() {
        return base;
    }

    public double getAltura() {
        return altura;
    }

    @Override
    public double calcularArea() {
        return base * altura;
    }
}

File: android/madrid/cesjuanpablosegundo/referencia-estilo/p2-figuras/model/Rectangulo.kt
package model

class Rectangulo(var base: Double, var altura: Double) : Figura() {

    fun calcularPerimetro(): Double {
        return 2 * (base + altura)
    }

    override fun calcularArea(): Double {
        return base * altura
    }

    override fun mostrarDatos() {
        println("Tipo = Rectangulo")
        println("Base = $base")
        println("Altura = $altura")
        println("Perimetro = ${calcularPerimetro()}")
        super.mostrarDatos()
    }
}
